dist/tools/firmware_metadata/README.md

```diff
@@ -0,0 +1,33 @@
+# Metadata generator for OTA firmware updates
+This program will generate a binary file containing a metadata structure as
+follows:
+
+```c
+typedef struct FW_metadata {
```

Comment:
Oh yeah, I got it. Well, this approach is mainly intended for cortex-m3 cpus, since I'm relying on the fact that the vector table can be stored in any 256 byte aligned place on the ROM, which I don't know it would work for 16 or 8 bit platforms.